tests/test_client.py

```diff
@@ -0,0 +1,442 @@
+from click import group, pass_context, version_option
+from click.testing import CliRunner
+from requests_mock import Adapter
+from re import compile, sub
+from unittest import TestCase
+
+from pureport.api.client import Client
+from pureport.cli.cli import commands
+from pureport.cli.util import construct_commands
+from pureport.util.api import PureportSession
+
+
+def __create_mock_client():
+    """
+    This creates a mock :class:`Client` instance.  It uses
+    a real client to call out to the server for the OpenAPI data
+    and constructs mock paths that can emulate the resources available
+    on the server
+    :rtype: Client
+    """
+    adapter = Adapter()
+    # Loads all paths from OpenAPI as mock path matchers
+    # The paths come in the form '/accounts/{account_id}', which we'll
+    # instead convert to proper Regex path's /accounts/([^/]+)
+    open_api = Client().open_api()
+    for path, path_obj in open_api['paths'].items():
+        url = compile('^mock://' + sub(r'{[^}]+}', '([^/]+)', path) + '$')
+        for method, method_obj in path_obj.items():
+            adapter.register_uri(method.upper(), url, json={})
+    adapter.register_uri('POST',
+                         'mock:///login',
+                         json={
+                             'access_token': '',
+                             'refresh_token': '',
+                             'expires_in': 2 ^ 31
+                         })
+    session = PureportSession(base_url='mock://')
+    session.mount('mock', adapter)
+    return Client(base_url='mock://', key='', secret='', session=session)
+
+
+def __create_mock_cli(pureport_client):
+    """
+    This creates a mock CLI using a mock client.
+    :param Client pureport_client:
+    :rtype: click.Command
+    """
+    @group(context_settings={'auto_envvar_prefix': 'PUREPORT'})
+    @version_option()
+    @pass_context
+    def cli(ctx):
+        """
+        \f
```

Comment:
Just a question. 
I've seen this `\f` a few times.
What is this for?

Reply:
So click, the cli tool, read's the pydoc off the function and uses that to display the help text in the CLI.  It uses the `\f` to separate what you want to show from what you don't want to show.  In most cases, we don't want to show the part of the pydoc that describe the parameter types, it just looks weird.

https://click.palletsprojects.com/en/7.x/documentation/#truncating-help-texts